File: DiceGame/DiceGame_typescript/Dice.ts
module DiceGame_TS {
    export class Dice {
        private m_value: number;

        constructor() {
            this.m_value = 0;
        }

        public Roll(): void {
            this.m_value = Math.floor(Math.random() * (7 - 1)) + 1;
        }

        public GetValue(): number {
            return this.m_value;
        }
    }
}

File: DiceGame/DiceGame_Swift1.2/DiceGame_Swift1.2/Dice.swift
//
//  Dice.swift
//  Dice
//
//  Created by Andreas Bom on 2015-09-02.
//  Copyright (c) 2015 Andreas Bom. All rights reserved.
//

import Foundation


class Dice {
    var m_value:Int
    
    init(){
        m_value = 0
    }
    
    func Roll(){
        m_value = Int(arc4random_uniform(5) + 1)
        
    }
    
    func GetValue()->Int{
        return m_value
    }
}